activerecord/test/cases/encryption/envelope_encryption_key_provider_test.rb

```diff
@@ -29,6 +29,15 @@ class ActiveRecord::Encryption::EnvelopeEncryptionKeyProviderTest < ActiveRecord
     assert_equal key.secret, @key_provider.decryption_keys(encrypted_message).first.secret
   end
 
+  test "decryption_keys raises Errors::Decryption if encrypted message has no DEK" do
```

Comment:
I don't think this tests demonstrate the problem you are trying to fix. Can you write a test that assets that the message will be decrypted by the previous provider automatically?

Reply:
Just to make clear, keep this test, but let's add a new test to make sure that raising `Errors::Decryption` actually allows the message to be decrypted.

Reply:
@rafaelfranca hey! Thanks for the review!
Did as you suggested - can you take a look please?